manifests/dropin_file.pp

```diff
@@ -69,11 +69,20 @@
     show_diff               => $show_diff,
   }
 
+  exec { "${name}-dropin-systemctl-daemon-reload":
+    command     => 'systemctl daemon-reload',
+    refreshonly => true,
```

Comment:
Rather than `refreshonly` I wonder if `onlyif => 'systemctl show -p NeedDaemonReload | grep yes'` would work.

Reply:
Seems like a good idea :thinking:

Reply:
Needs the unit name in there of course.
```
onlyif => 'systemctl show -p NeedDaemonReload $unit | grep yes'
```
The downside is this being run potentially many times per unit per puppet run always but the simplicity wins for me. 

`systemd::unit_file` would need the same code for an unmanned .service and .timer so would be nice to really only have one exec per unit. As mentioned else where - defined type and `ensure_resource`

Reply:
The `onlyif` incurs overhead for every run. Probably a personal preference, but `refreshonly` coupled with `subscribe` feels simple and self descriptive to me.

Reply:
The `onlyif => "systemctl show -p NeedDaemonReload $unit | grep yes"` way makes sense if for some reason a user has both a `systemd::unit_file` and a `systemd::dropin_file` for the same service.

I consider this to be an edge case because if you deploy the unit, why bother with a dropin? However, if you do need this unusual setup, it will work but `systemctl daemon-reload` will be called twice. The overhead of that second extraneous call that happens only on changes is probably not too far off from the `NeedDaemonReload` test it's avoiding for every single runs.